### PR TITLE
Update base add docker compose yml

### PR DESCRIPTION
**Pull request by taniwallach**
The `docker-compose.yml` will make startup/shutdown easier for Docker users.
It also provides persistent storage for the SQLite database and an easy approach to using "bind" mounts.
Dockerfile was updated to use Ubuntu focal (20.04 LTS) instead of Xenial (16.04 LTS) and to use Node.js v14 instead of the obsolete v12.

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @taniwallach! I've created [OSPR-5380](https://openedx.atlassian.net/browse/OSPR-5380) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.


:warning: We can't start reviewing your pull request until you've submitted a     [signed contributor agreement](https://open.edx.org/wp-content/uploads/2019/01/individual-contributor-agreement.pdf)     or indicated your institutional affiliation. Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.

**Comment by natabene** [comment]
@taniwallach Thank you for joining the Open edX community! We have added your user to our records and can now review your code going forward.

**Comment by taniwallach** [comment]
@natabene -Thanks. Happy to be able to contribute.

File: docker-compose.yml
Comment (symbolist) [inline docker-compose.yml:6]: Can you please remove all the unneeded commented out code?

File: docker-compose.yml
Comment (symbolist) [inline docker-compose.yml:4]: Optional: Can add `container_name: edx.devstack.xblock-sdk` and then use it instead of the auto-generated name `xblock-sdk_xblock_sdk_1 `.

File: README.rst
Comment (symbolist) [inline README.rst:82]: I think it would be better if instead of this, at line 71 we added "If using docker, all these commands need to be run inside the xblock-sdk container."?

File: Dockerfile
Comment (symbolist) [inline Dockerfile:28]: I already have some container using `8000` so wasn't able to test this. 🙂  Since `8000` is the default port for Django maybe we can change this to something which has less chance of a conflict, like `18340`?

File: README.rst
Comment (symbolist) [inline README.rst:61]: Can you put `#!/usr/bin/env python3` at the top of `manage.py`? That way we don't have to keep prepending `python` when running any commands in the shell.

Currently it leads to:
```
root@localhost:/usr/local/src/xblock-sdk# ./manage.py
./manage.py: line 1: Manage.py file for XBlock: command not found
./manage.py: line 2: import: command not found
./manage.py: line 3: import: command not found
./manage.py: line 5: import: command not found
./manage.py: line 8: syntax error near unexpected token `"DJANGO_SETTINGS_MODULE",'
./manage.py: line 8: `    os.environ.setdefault("DJANGO_SETTINGS_MODULE", "workbench.settings")'
```

**Review by symbolist — CHANGES_REQUESTED** [review]
Hi @taniwallach, I did a pass through the PR. Overall looking good; have left a few comments. 🙂

**Comment by symbolist** [comment]
@natabene Since I am CC on the repo I took the liberty to go ahead and do a review pass (but let me know if there is some process for selecting the reviewer). 🙂

File: Dockerfile
Comment (taniwallach) [reply Dockerfile:28]: XBlock SDK uses port 8000 by default. I'm not sure I should be changing this.
I'll look at all the other comments more carefully next week.

**Comment by natabene** [comment]
@symbolist Thank you so much, please do review and merge when ready. CC @sarina

**Comment by natabene** [comment]
@taniwallach Please let me know once you have implemented the feedback, so we can review again.

File: README.rst
Comment (taniwallach) [reply README.rst:23]: Yes. Being fixed.

File: docker-compose.yml
Comment (taniwallach) [reply docker-compose.yml]: Done

File: README.rst
Comment (taniwallach) [reply README.rst]: Great suggestion. Done.

File: README.rst
Comment (taniwallach) [reply README.rst]: Done.

File: README.rst
Comment (taniwallach) [reply README.rst:51]: Done. I also added a few of the other make targets shown in the sample.

File: Dockerfile
Comment (taniwallach) [reply Dockerfile:28]: I'm not changing the port. I don't feel qualified to determine if such a change is appropriate.
Since the xblock-sdk was intended to provide a minimal framework for XBlock development, it was not intended to be a piece which integrates with a full dev-stack, but as an alternative.
Prior developers would be more qualified to make the call on whether changing the port number.

File: docker-compose.yml
Comment (taniwallach) [reply docker-compose.yml:4]: Done

**Comment by taniwallach** [comment]
@symbolist  - Thanks for all the feedback. I did my best to implement the necessary changes and suggestions, except with regard to the port number being used. See above for why I am not handling that suggestion. The revised version and the new make targets seems to all seems to work.
@natabene - I think this is ready for a second review. Sorry that I could not handle this sooner.

**Comment by natabene** [comment]
@symbolist Can you please review this as a Core Committer? If you cannot, please let me know and I will find another reviewer.

**Comment by symbolist** [comment]
@natabene @taniwallach Yup, I'll be able to review this week!

File: README.rst
Comment (symbolist) [inline README.rst:63]: Since this deletes state, in general use `stop` is what is preferred so let's change to that.

File: README.rst
Comment (symbolist) [inline README.rst:80]: Can you also add a make command `dev.shell` for this?

File: Dockerfile
Comment (symbolist) [reply Dockerfile:28]: Hmm, when I try to run `./manage runserver` inside the container I am still getting `Error: That port is already in use.`. How are you working with this?

**Review by symbolist — CHANGES_REQUESTED** [review]

File: Dockerfile
Comment (taniwallach) [reply Dockerfile:28]: The XBlock SDK is meant to be used for development without a running devstack, etc.
Since that is the intended usage - the conflict is not supposed to occur.
I'm new to this, and am not taking the initiative to change to port number, someone else should consider if that is suitable and handle it in a different PR.

File: README.rst
Comment (taniwallach) [reply README.rst:71]: Done, with an addition about `down` in addition to `stop`.

File: README.rst
Comment (taniwallach) [reply README.rst:94]: I added it as `app.shell` to match the sample in https://github.com/edx/edx-cookiecutters/blob/master/cookiecutter-django-ida/%7B%7Bcookiecutter.repo_name%7D%7D/Makefile

**Comment by taniwallach** [comment]
About the use of port 8000, which will conflict with other Django applications, and with the edX devstack: the use of XBlock-SDK is intended to be separate of the use of devstack, and that is apparently why the developers decided to use the standard Django port (8000) for the SDK. Since considering whether this should be changed is a more significant change than the updates in this PR, if you think this is important, it should be handled by a different PR. I recommend consulting with someone with more experience with this tool before making a change.

File: Dockerfile
Comment (symbolist) [reply Dockerfile:28]: Hmm, actually I tested this after restarting docker (so nothing else was running) which is why I asked. Let me look into this some more to see why it is not working for me.

Comment (taniwallach) [reply Dockerfile:28]: Sorry. I did not explain well.

As soon as I run `make dev.up` the expected server is available on `http://localhost:8000/`.

---

If I enter a shell using `make app-shell` and in the shell run
```
./manage.py runserver
```
I also get an error - as it is already running.
```
root@localhost:/usr/local/src/xblock-sdk# ./manage.py runserver
Performing system checks...

System check identified no issues (0 silenced).
February 25, 2021 - 17:16:01
Django version 2.2.14, using settings 'workbench.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CONTROL-C.
Error: That port is already in use.

```

Comment (symbolist) [reply Dockerfile:28]: Ah, I missed the `CMD` specified below. Great, it is working as expected!

**Comment by symbolist** [comment]
@taniwallach Thanks! Can you squash all your commits and add a nice commit message? Then I'll merge. 🙂

**Comment by taniwallach** [comment]
> Thanks! Can you squash all your commits and add a nice commit message? Then I'll merge.

@symbolist Thanks for all the review work and suggestions. All squashed into a single commit (`git rebase -i` and a `git push --force`) which is what I think you wanted done to clean it up nicely.

**Review by symbolist — APPROVED** [review]
- [x] I tested this by provisioning and destroying the stack and testing the server and tests run as expected.
- [x] I read through the code
- [x] Includes documentation

**Comment by edx-webhook** [comment]
@taniwallach 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/xblock-sdk&entry.867055334=https://github.com/edx/xblock-sdk/pull/194&entry.1484655318=https://github.com/taniwallach&entry.752974735=2021-01-20+13:02&entry.1917517419=2021-03-01+14:25&entry.2133058324=Yes) so we can improve your experience in the future.

**Comment by taniwallach** [comment]
@symbolist Thanks for all the helpful feedback and the careful review.

**Pull request merged**
